codalab/apps/web/utils.py

```diff
@@ -118,3 +119,236 @@ def inheritors(klass):
+    if not url or url == '':
+        logger.error("Received an invalid url to convert to a key: {}".format(url))
+        return
+    # Remove the beginning of the URL (before bucket name) so we just have the path(key) to the file
+    key = url.split("{}/".format(settings.AWS_STORAGE_PRIVATE_BUCKET_NAME))[-1]
+    # Path could also be in a format <bucket>.<url> so check that as well
+    key = key.split("{}.{}/".format(settings.AWS_STORAGE_PRIVATE_BUCKET_NAME, settings.AWS_S3_HOST))[-1]
+    return key
+
+
+def get_competition_size_data(competition):
+    data = {
+        'id': competition.id,
+        'title': competition.title,
+        'creator': "{0} ({1})".format(competition.creator.email, competition.creator.username),
+        'is_active': competition.is_active,
+        'submissions': 0,
+        'datasets': 0,
+        'dumps': 0,
+        'bundle': 0,
+    }
+    data_file_attrs = [
+        'reference_data_organizer_dataset',
+        'input_data_organizer_dataset',
+        'scoring_program_organizer_dataset',
+        'public_data_organizer_dataset',
+        'starting_kit_organizer_dataset',
+        'ingestion_program_organizer_dataset',
+    ]
+    keys_to_total = [
+        'submissions',
+        'datasets',
+        'bundle',
+        'dumps',
+    ]
+    for phase in competition.phases.all():
+        # submissions
+        for submission in phase.submissions.all():
+            # submission.size should call get_submission_size anyway.
+            data['submissions'] += submission.size
+        # datasets
+        for attr in data_file_attrs:
+            dataset = getattr(phase, attr)
+            if dataset:
+                data['datasets'] += dataset.size
+    # bundle
+    if hasattr(competition, 'bundle'):
+        data['bundle'] = competition.bundle.size
+    # dumps
+    if competition.dumps.exists():
+        data['dumps'] = sum([dump.size for dump in competition.dumps.all()])
+    data['total'] = sum([data[key] for key in keys_to_total])
+    return data
+
+def get_submission_size(submission):
+    total = 0
+    if not os.environ.get('PYTEST'):
+        file_attrs = [
+            'inputfile',
+            'runfile',
+            'output_file',
+            'private_output_file',
+            'stdout_file',
+            'stderr_file',
+            'history_file',
+            'scores_file',
+            'coopetition_file',
+            'detailed_results_file',
+            'prediction_runfile',
+            'prediction_output_file',
+            'prediction_stdout_file',
+            'prediction_stderr_file',
+            'ingestion_program_stdout_file',
+            'ingestion_program_stderr_file',
+        ]
+        total += get_filefield_size(submission, 'file', aws_attr='s3_file', s3direct=True)
+        for file_attr in file_attrs:
+            total += get_filefield_size(submission, file_attr)
+    return total
+
+def get_filefield_size(obj, attr, aws_attr=None, s3direct=False):
+    size = None
```

Comment:
did you get a chance to test this vs azure or other storage types?

Reply:
I have not yet. This is on my todo.

I realized during writing most of this that I had made it very S3 specific and started working on trying to leverage Django Storage Classes instead when this was not the case.

Our Azure Storage Class does implement the required methods, and looks to operate on the same manner as S3 (IE: name being the path/key for the obj/file in storage).